driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1593,15 +1593,15 @@ private void connect() {
                 if (serverResponseVersion == attemptedVersion) {
                     // We're post CASSANDRA-11464: the server replies with the same version that the client tried (to
                     // ensure that the client can decode the response).
-                    // Retry with a lower version (if it's too low we'll eventually reach our min and fail).
+                    // Retry with a lower version (if it's too low we'll fail at the next attempt, as we only retry once).
```

Comment:
> In that case, what's the point of even retrying?

That was my point since the beginning. But @jorgebay convinced me: the fact that retrying at N-1 is necessarily going to fail is not a documented feature, but an implementation detail at this point. We don't know if this is going to always be the case, so retrying at N-1 is more future-proof.